### PR TITLE
Fix/duplicate competencies is allowed

### PR DESCRIPTION
**Pull request by YouQam**

### CODE REVIEW

File: src/app/issuer/components/badgeclass-edit-form/badgeclass-edit-form.component.ts [hunk 1]
```diff
@@ -1043,22 +1049,23 @@ export class BadgeClassEditFormComponent extends BaseAuthenticatedRoutableCompon
...
-        const hand = this.badgeClassForm.controls.competencies.value.
+        const storedCompetencies = this.badgeClassForm.controls.competencies.value.
             // Hand competencies get added automatically at submitting
             //filter(c => c.added).
             map(c => c.name);
-        const ai = this.badgeClassForm.controls.aiCompetencies.value.
-            filter(c => c.selected).
-            map((c,i) => this.aiCompetenciesSuggestions[i].preferred_label);
+		
+		const newSelectedCompetencies = this.badgeClassForm.controls.aiCompetencies.value
```
Comment (timber-they) [inline src/app/issuer/components/badgeclass-edit-form/badgeclass-edit-form.component.ts:1064]: Why did you rename the variables? I find the new names quite misleading, since the `storedCompetencies` are not (only) stored competencies, but also those selected by hand. Also, `newSelectedCompetencies` are not all new selected competencies, but only the AI competencies.

Comment (YouQam) [reply src/app/issuer/components/badgeclass-edit-form/badgeclass-edit-form.component.ts:1064]: I updated them as they were quite incomplete/misleading. you are right `storedCompetencies` includes those added by hand. I've updated both variables to be more meaningful.
